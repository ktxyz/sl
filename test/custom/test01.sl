integer age; 
string msg;
integer i; 
string name;

// Simulate input: age=25, name="Alice" 
age := readint; 
name := readstr; 

msg := concatenate("Name: ", name); 
print(msg); 

if age > 20 and length(name) > 3 then 
  begin 
    for i := 1 to age / 5 do 
      begin 
        print(concatenate("Decade ", substring("IIIII", 1, i))); 
        if i = 3 then break; 
      end; 
  end; 

// Force type error if uncommented 
// string error := age; 