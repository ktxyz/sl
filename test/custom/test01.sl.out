Name: Alice  
Decade I  
Decade II  
Decade III